### PR TITLE
test: add two e2e tests for pulsar source

### PR DESCRIPTION
**Pull request by Honeta**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

add e2e test for astra-streaming source and streamnative-cloud source

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]
basically LGTM, waiting for @huangjw806 review

**Review by huangjw806 — APPROVED** [review]
LGTM! Please merge after the test on the main-cron pipeline is successful.

**Pull request merged**
